File: java/queues/LinkedQueue.java
package queues;

/** Realization of a FIFO queue as an adaptation of a SinglyLinkedList. */

import fds.SinglyLinkedList;

public class LinkedQueue<E> implements Queue<E> {
    private SinglyLinkedList<E> list = new SinglyLinkedList<>(); // an empty list

    public LinkedQueue() { // new queue relies on empty list
    }

    @Override
    public E dequeue() {
        return list.removeFirst();
    }

    @Override
    public void enqueue(E e) {
        list.addFirst(e);
    }

    @Override
    public E first() {
        return list.first();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public int size() {
        return list.size();
    }

}

File: python/A First Course on Data Structures in Python/ds/queue/linkedqueue.py
from ..linkedlist import LinkedList


class LinkedQueue:
    def __init__(self):
        self._L = LinkedList()

    def enqueue(self, item):
        self._L.add_last(item)

    def dequeue(self):
        return self._L.remove_first()

    def peek(self):
        # or we have to use remove_first() and add_first() to get item at head
        return self._L.get_head()

    def __len__(self):
        return len(self._L)

    def is_empty(self):
        return len(self) == 0
